storybook/sectionsdecoratormodel.cpp

```diff
@@ -0,0 +1,162 @@
+#include "sectionsdecoratormodel.h"
+
+#include <QScopeGuard>
+
+SectionsDecoratorModel::SectionsDecoratorModel(QObject *parent)
+    : QAbstractListModel{parent}
+{
+}
+
+void SectionsDecoratorModel::setSourceModel(QAbstractItemModel *sourceModel)
+{
+    if (m_sourceModel != nullptr) {
+        qWarning("Changing source model is not supported!");
```

Comment:
::initialize and :: findSectionRole assume m_sourceModel != nullptr. Calling setSourceModel(nullptr) would crash the app.